packages/chat/src/components/VoiceWidget/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { useState } from 'react';
+
+import { Button } from '../Button';
+import { ButtonIcon } from '../Button/ButtonIcon';
+import { ButtonVariant } from '../Button/constants';
+import { BottomLinks } from '../NewFooter/BottomLinks';
+import useMicrophoneAmplitude from './use-microphone-amplitude';
+import {
+  buttonContent,
+  buttonModifier,
+  buttonText,
+  circle,
+  controlSection,
+  imageStyles,
+  linkSectionModifier,
+  titleStyle,
+  voiceWidgetContainer,
+  voiceWrapper,
+} from './VoiceWidget.css';
+
+interface IVoiceWidget {
+  imageSrc: string;
+  isListening?: boolean;
+  isTalking?: boolean;
+  variant: 'full' | 'expanded' | 'compact';
+  footer: {
+    extraLinkText: string;
+    extraLinkUrl: string;
+    showPoweredBy: boolean;
+  };
+}
+
+export const VoiceWidget: React.FC<IVoiceWidget> = ({ imageSrc, variant = 'full', isListening, isTalking, footer }) => {
+  const [isCalling, setIsCalling] = useState(false);
+  const { showPoweredBy, extraLinkText, extraLinkUrl } = footer;
+
+  const amplitude = useMicrophoneAmplitude();
+
+  const handleButtonClick = () => {
+    setIsCalling((prev) => !prev);
+    if (isCalling) {
```

Comment:
In react (from what I know) we don't have a guarantee as to when a 'set state' method will change the state. (It should happen instantaneously, but sometimes for performance reasons it might not).

So, I would place this `if` before the `setIsCalling()` method, so it's definitive.